### PR TITLE
Improve ValueSource javadoc a bit

### PR DESCRIPTION
**Pull request by mlopatkin**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test BD

File: subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java [hunk 1]
```diff
@@ -45,15 +45,20 @@
...
- * It is possible to have some Gradle services to be injected into the implementation, similar to
- * tasks and plugins. It can be done by adding a parameter to the constructor and annotating the
+ * It is possible to have some Gradle services to be <a href="https://docs.gradle.org/current/userguide/custom_gradle_types.html#service_injection">injected</a>
+ * into the implementation, similar to tasks and plugins.
+ * It can be done by adding a parameter to the constructor and annotating the
  * constructor with the {@code @Inject} annotation:
  * <pre>
- * public class MyValueSource implements ValueSource&lt;...&gt; {
- *     &#064;Inject
+ * public abstract class MyValueSource implements ValueSource&lt;String, ValueSourceParameters.None&gt; {
```
Comment (bamboo) [inline subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java:53]: 👍

File: subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java [hunk 2]
```diff
@@ -45,15 +45,20 @@
...
+ *     private final ExecOperations execOperations;
+ *
+ *     {@literal @}Inject
  *     public MyValueSource(ExecOperations execOperations) {
- *         ...
+ *         this.execOperations = execOperations;
```
Comment (bamboo) [inline subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java:58]: 👍

**Review by bamboo — APPROVED** [review]

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**